### DISCUSSION TITLE
Configuring SSO_ONLY and SSO_MASTER_PASSWORD_POLICY

### DISCUSSION THREAD

**Discussion by magajimenez10** [dani-garcia/vaultwarden#5199] [Q&A]
Hi! I’m testing the two configuration options, SSO_ONLY and SSO_MASTER_PASSWORD_POLICY. I would like to configure it so that login can only be done via SSO and that the master password is not required. Alternatively, if the master password is not implemented, I want to always require SSO login, even if the user tries to use the master password. I thought this was the behavior that SSO_ONLY triggers, but I’m still able to log in with the master password without SSO, even when SSO_ONLY is set to true.

Thank you very much!

**Comment by momonator25** [comment]
Do you have found a solution for this? I'm at the same point. SSO with Authelia is working but it still wants the Master Password.
Setting `SSO_MASTER_PASSWORD_POLICY=false` does result in `Error loading config:
  Invalid sso_master_password_policy: parsed value is not a JSON object` but I have not found any info how the structure should be.

**Reply by stefan0xC** [reply]
That's not how SSO in Bitwarden or Vaultwarden works or what the master password policy is for. Cf. https://bitwarden.com/help/sso-decryption-options/ and https://bitwarden.com/help/policies/#master-password-requirements and https://github.com/dani-garcia/vaultwarden/discussions/6683#discussioncomment-15445549

**Reply by momonator25** [reply]
Got it! Thanks for the fast reply this saves me further investigation as **it's not easily possible to disable Master-Password for SSO** from what i have read in your links.

**Reply by stefan0xC** [reply]
Yeah. SSO just decouples authentication from decryption and Vaultwarden does not support setting a different decryption option at the moment (and likely not any time soon). edit: I mean we support login with device just not with SSO (but like I said in the other thread someone would have to look into that).
